opencga-app/src/main/java/org/opencb/opencga/app/migrations/v2_4_2/catalog/RecoverProbandSamplesInCases.java

```diff
@@ -0,0 +1,164 @@
+package org.opencb.opencga.app.migrations.v2_4_2.catalog;
+
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.Sorts;
+import org.apache.commons.collections4.CollectionUtils;
+import org.bson.Document;
+import org.opencb.commons.datastore.core.ObjectMap;
+import org.opencb.commons.datastore.core.QueryOptions;
+import org.opencb.opencga.catalog.db.api.ClinicalAnalysisDBAdaptor;
+import org.opencb.opencga.catalog.db.api.IndividualDBAdaptor;
+import org.opencb.opencga.catalog.db.mongodb.MongoDBAdaptorFactory;
+import org.opencb.opencga.catalog.db.mongodb.converters.ClinicalAnalysisConverter;
+import org.opencb.opencga.catalog.exceptions.CatalogException;
+import org.opencb.opencga.catalog.migration.Migration;
+import org.opencb.opencga.catalog.migration.MigrationTool;
+import org.opencb.opencga.core.models.clinical.ClinicalAnalysis;
+import org.opencb.opencga.core.models.common.Enums;
+import org.opencb.opencga.core.models.individual.Individual;
+import org.opencb.opencga.core.models.sample.Sample;
+
+import java.util.*;
+
+@Migration(id = "recover_proband_samples_in_cases_TASK-1470",
+        description = "Recover lost samples in clinical collection #TASK-1470", version = "2.4.2",
+        language = Migration.MigrationLanguage.JAVA,
+        domain = Migration.MigrationDomain.CATALOG,
+        date = 20220725)
+public class RecoverProbandSamplesInCases extends MigrationTool {
+
+    @Override
+    protected void run() throws Exception {
+        MongoCollection<Document> audit = getMongoCollection(MongoDBAdaptorFactory.AUDIT_COLLECTION);
+        ClinicalAnalysisConverter converter = new ClinicalAnalysisConverter();
+
+        queryMongo(MongoDBAdaptorFactory.CLINICAL_ANALYSIS_COLLECTION,
+                new Document()
+                        .append("proband.samples", null)
+                        .append("locked", false),
```

Comment:
Doesn't this affect to any locked clinical analysis?

Reply:
Actually, it could... I will change that